component/src/main/java/org/wso2/carbon/identity/authenticator/nuxeo/NuxeoAuthenticator.java

```diff
@@ -0,0 +1,386 @@
+     * This method is overridden to check additional condition like whether request is having
+     * sendToken, token parameters, generateNuxeoToken and authentication name.
+     *
+     * @param request  Http servlet request
+     * @param response Http servlet response
+     * @param context  AuthenticationContext
+     * @throws AuthenticationFailedException User tenant domain mismatch
+     */
+    @Override
+    protected void processAuthenticationResponse(HttpServletRequest request, HttpServletResponse response,
+                                                 AuthenticationContext context) throws AuthenticationFailedException {
+
+        String clientId = null;
+        try {
+            Map<String, String> authenticatorProperties = context.getAuthenticatorProperties();
+            clientId = authenticatorProperties.get(OIDCAuthenticatorConstants.CLIENT_ID);
+            String clientSecret = authenticatorProperties.get(OIDCAuthenticatorConstants.CLIENT_SECRET);
+            String tokenEndPoint = getTokenEndpoint(authenticatorProperties);
+            String callbackUrl = getCallbackUrl(authenticatorProperties);
+            String userInfoUrl = getUserInfoEndpoint(authenticatorProperties);
+            OAuthAuthzResponse authorizationResponse = OAuthAuthzResponse.oauthCodeAuthzResponse(request);
+            String code = authorizationResponse.getCode();
+            AuthenticatedUser authenticatedUserObj;
+            Map<ClaimMapping, String> claims;
+            OAuthClientRequest accessRequest = getAccessRequest(tokenEndPoint, clientId, code, clientSecret,
+                    callbackUrl);
+            OAuthClient oAuthClient = new OAuthClient(new URLConnectionClient());
+            OAuthClientResponse oAuthResponse = getResponseOauth(oAuthClient, accessRequest);
+            String accessToken = oAuthResponse.getParam(OIDCAuthenticatorConstants.ACCESS_TOKEN);
+            String loggedInUserInfo = getUserInfoJson(userInfoUrl, accessToken);
+            if (loggedInUserInfo.isEmpty()) {
+                throw new AuthenticationFailedException("Getting empty or null value for logged in user info from " +
+                        "a user info endpoint: " + userInfoUrl);
+            }
+            Object loggedInUserID = JSONUtils.parseJSON(loggedInUserInfo).get(NuxeoAuthenticatorConstants.
+                    LOGGED_IN_USER_IDENTIFIER);
+            claims = mapUserInfo(loggedInUserInfo);
+            authenticatedUserObj = AuthenticatedUser.createFederateAuthenticatedUserFromSubjectIdentifier(
+                    String.valueOf(loggedInUserID));
+            authenticatedUserObj.setAuthenticatedSubjectIdentifier(String.valueOf(loggedInUserID));
+            authenticatedUserObj.setUserAttributes(claims);
+            context.setSubject(authenticatedUserObj);
+        } catch (OAuthProblemException e) {
+            throw new AuthenticationFailedException("Exception while getting the oAuth code from HttpServletRequest "
```

Comment:
Is this for authorization code or access token ? Which ever it is, better to mention specifically. Also, no need to mention about HttpServletRequest

Reply:
This exception thrown from (https://github.com/wso2-extensions/identity-outbound-auth-nuxeo/pull/1/commits/7f4b5ef07a3874ddd8c8aa9d0b0a6f89cb4a2c56#diff-eb4ac1c1455492b50499f6482aceb4baR200) . So This is for authorization code.